odoo/addons/base/models/ir_mail_server.py

```diff
@@ -556,6 +591,80 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.model
+    def _find_mail_server(self, mail_from):
```

Comment:
Could be great to have it working for multiple emails. Would return a list of tuple (mail_server_id, smtp_from, email_from) or {'email_from': ..., 'smtp_from': ..., 'mail_server_id': ...} . Then could be used in mail_mail.split_by_server to optimize it :D .